app/src/main/java/org/mozilla/tv/firefox/utils/ServiceLocator.kt

```diff
@@ -38,6 +41,12 @@ import org.mozilla.tv.firefox.pocket.Pocket
  *   open val telemetry: TelemetryInterface by lazy { SentryWrapper() }
  *   ```
  */
-open class ServiceLocator {
-    open val pocket by lazy { Pocket() }
+class ServiceLocator {
+    private val pocketEndpoint by lazy { PocketEndpoint }
+    private val buildConfigDerivables by lazy { BuildConfigDerivables() }
+    private val pocketRepoStateMachine by lazy { PocketRepoStateMachine() }
+    val pocketRepo = PocketRepo(pocketEndpoint, pocketRepoStateMachine, buildConfigDerivables).apply {
```

Comment:
Do you think all of pocket related instances need to be exposed here? I understand for testability it makes sense to pass them into PocketRepo's constructors since we can mock the dependencies easier. But I even wonder if some of these instance memory need to be allocated for the entire lifecycle of the application

Reply:
> Do you think all of pocket related instances need to be exposed here?

I think it makes sense to do object setup in the `ServiceLocator`.  They aren't part of its public API, so I wouldn't consider them exposed.  If setup gets big we can break these into individual factories (and if it gets huge we can introduce a framework), but that's a pretty easy refactor so I would prefer to keep it simple until we need to.

> But I even wonder if some of these instance memory need to be allocated for the entire lifecycle of the application

Good point.  A lot of those 'by lazy's are useless, I'll pull them out.